pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -392,7 +392,11 @@ func (c *Controller) onServiceEvent(curr interface{}, event model.Event) error {
 		delete(c.networkGateways, svcConv.Hostname)
 		c.Unlock()
 	default:
-		if isNodePortGatewayService(svc) {
+		// First, process nodePort gateway service, whose externalIPs specified
+		// and loadbalancer gateway service
+		if svcConv.Attributes.ClusterExternalAddresses != nil {
```

Comment:
Will this make it so we never update the service node port address after we get the first one?

Reply:
if the externalIPs is specified, then nodeport is not respected. This is to make this a determined behavior.

Reply:
I didn't catch that `svcConv` was freshly generated here, I thought externalIPs may have already been populated by NodePort on a previous update. 

One other question: When using the external IPs we won't map the node port to the service port. Do we expect that services with type NodePort will also be configured to expose the Service port (usually 15443)?

Reply:
Yes, in this case remote cluster's gateway can be accessed by externalIP:15443

Reply:
@stevenctl Any other concern

Reply:
No, this looks good. Thanks for the clarifications!